### PR TITLE
 Added yii\rbac\ManagerInterface::getRolesByRole($roleName) method

### PR DESCRIPTION
**Pull request by githubjeka**
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | yes |
| Breaks BC? | yes |
| Fixed issues | #11245 |

P.S.
Tests pass:

``` php
public function testGetRolesByRole()
{
        //...
        $roles = $this->auth->getRolesByRole('author');
        $this->assertCount(2, $roles);
        $this->assertArrayHasKey('author', $roles);
        $this->assertArrayHasKey('reader', $roles);

        $roles = $this->auth->getRolesByRole('admin');
        $this->assertCount(3, $roles);
        $this->assertArrayHasKey('admin', $roles);
        $this->assertArrayHasKey('author', $roles);
        $this->assertArrayHasKey('reader', $roles);
    }
```

image

### CODE REVIEW

File: framework/UPGRADE.md [hunk 1]
```diff
@@ -15,6 +15,12 @@ php composer.phar self-update
 php composer.phar global require "fxp/composer-asset-plugin:^1.2.0"
 ```
 
+Upgrade from Yii 2.0.9
+----------------------
+
+* RBAC: Interface `ManagerInterface` has been changed to added `getRolesByRole()` method. If you create your Manager via
```
Comment (samdark) [inline framework/UPGRADE.md]: `getRolesByRole()` method was added to `ManagerInterface`. If you've implemented your own RBAC manager you need to implement new method.

File: framework/rbac/DbManager.php [hunk 2]
```diff
@@ -473,6 +473,29 @@ public function getRolesByUser($userId)
     /**
      * @inheritdoc
      */
+    public function getRolesByRole($roleName)
```
Comment (samdark) [inline framework/rbac/DbManager.php]: Could it be named better? `getChildRoles`?

File: framework/rbac/DbManager.php [hunk 3]
```diff
@@ -473,6 +473,29 @@ public function getRolesByUser($userId)
...
+    {
+        $role = $this->getRole($roleName);
+
+        if (is_null($role)) {
+            throw new InvalidParamException("Role '$roleName' not found.");
```
Comment (samdark) [inline framework/rbac/DbManager.php]: Usually `"` are used in string messages, not `'`.

**Review by samdark — CHANGES_REQUESTED** [review]

File: framework/rbac/DbManager.php (hunk 2 above)
Comment (SilverFire) [inline framework/rbac/DbManager.php]: `getChildRoles` is better

**Comment by githubjeka** [comment]
Renamed. Test pass.

**Review by samdark — APPROVED** [review]

**Comment by klimov-paul** [comment]
This is a BC break, since extra method introduced to `ManagerInterface`: any custom Manager class will cause PHP fatal error on framework update.

Either change should be reverted or UPGRADE note should be added.

**Comment by githubjeka** [comment]
Breaks BC?	yes

**Comment by klimov-paul** [comment]
Ah, UPDGADE note is already present - fine.

**Pull request merged**